application/models/User.php

```diff
@@ -148,6 +148,35 @@ public function attributeLabels()
         ];
     }
 
+    /**
+     * @inheritDoc
+     * Delete user in related model after deletion
+     * return void
+     **/
+    protected function afterDelete()
+    {
+        parent::afterDelete();
```

Comment:
As this are dependant/child relations, should this happen after delete or before delete?
Shouldn't we take advantage of relations in the model?

Reply:
after delete clearly

For the other : it work like that.

Reply:
1. Doing this afterDelete is clear (plugin can disable deletion for example)
2. Unsure we can do $this-> since user don't exist (to check)
3. We don't use relation in https://github.com/LimeSurvey/LimeSurvey/blob/dc083bd9e3280fbe58da885f157c6e48707cae2f/application/models/Survey.php#L324